### ISSUE TITLE
Mongo::Struct constructor_type is broken

### ISSUE THREAD

**Issue by v-kolesnikov** [rom-rb/rom-mongo#24]
Hi, folks!

I assume `ROM::Mongo::Struct` was broken after this commit: https://github.com/dry-rb/dry-struct/pull/55/files

https://github.com/rom-rb/rom-mongo/blob/cc335a47ee8504eb9ed660e130daea2557a235b8/lib/rom/mongo/struct.rb#L4-L8

I think we can just remove this option and  inherit `:permissive` option from ROM::Struct.
I would love to make PR as well.

@solnic @GustavoCaso

**Comment by GustavoCaso** [comment]
Yes @v-kolesnikov with the latest version of `dry-struct` this will raise an error. 

I haven't use `rom-mongo` ever but I think what you are suggesting is a good idea.
